### PR TITLE
Feature/auth frontend v2

### PR DESCRIPTION
**Pull request by IvanRuskevych**

### CODE REVIEW

File: packages/frontend/src/services/http.service.ts [hunk 1]
```diff
@@ -56,6 +56,7 @@ export class HttpService {
 				...this.populateTokenToHeaderConfig(),
 			};
 		}
+		console.log('config.data', config.data);
```
Comment (malyshkoserhii) [inline packages/frontend/src/services/http.service.ts:59]: remove log

File: packages/frontend/src/shared/components/Layout/Layout.tsx [hunk 2]
```diff
@@ -8,8 +8,36 @@ import {
...
+	const location = useLocation();
+	const navigate = useNavigate();
+	const token = new URLSearchParams(location.search).get('token');
+
+	useEffect(() => {
+		const verifyEmailToken = async (): Promise<void> => {
+			if (token) {
+				try {
+					const response = await verifyEmail(token);
+					console.log('verifyEmailToken response:', response);
```
Comment (malyshkoserhii) [inline packages/frontend/src/shared/components/Layout/Layout.tsx:26]: remove log

File: packages/frontend/src/shared/components/VerifyEmail/VerifyEmail.tsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+const authService = new AuthService();
+
+export const VerifyEmail: React.FC = () => {
+	const navigate = useNavigate();
+	const { token } = useParams();
+	const { loading } = useAuthStore();
+
+	useEffect(() => {
+		console.log('start useEffect');
```
Comment (malyshkoserhii) [inline packages/frontend/src/shared/components/VerifyEmail/VerifyEmail.tsx:18]: remove

File: packages/frontend/src/shared/components/VerifyEmail/VerifyEmail.tsx [hunk 4]
```diff
@@ -0,0 +1,45 @@
+import React, { useEffect } from 'react';
+import { useNavigate, useParams } from 'react-router-dom';
+import { toast } from 'react-toastify';
+
+import { AuthService } from '~services/auth.service';
+import { Loader, StyledNavLink } from '~shared/components';
+import { ROUTER_KEYS } from '~shared/keys';
+import { useAuthStore } from '~store/auth.store';
+
+const authService = new AuthService();
```
Comment (malyshkoserhii) [inline packages/frontend/src/shared/components/VerifyEmail/VerifyEmail.tsx:10]: export instance of this class from your service

`export const authService = new AuthService();`

File: packages/frontend/src/modules/AuthForms/RegisterForm.tsx [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+import { CustomForm, CustomField, Loader } from '~shared/components';
+
+export const RegisterForm: React.FC = () => {
+	const navigate = useNavigate();
+	const { user, register, loading } = useAuthStore();
+	const initialValues: User = {
+		username: user?.username || '',
+		email: user?.email || '',
+		password: user?.password || '',
+	};
```
Comment (malyshkoserhii) [inline packages/frontend/src/modules/AuthForms/RegisterForm.tsx:17]: useMemo

File: packages/frontend/src/modules/AuthForms/RegisterForm.tsx [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
+	const handleSubmit = async (
+		values: User,
+		{ resetForm }: { resetForm: () => void },
```
Comment (malyshkoserhii) [inline packages/frontend/src/modules/AuthForms/RegisterForm.tsx:21]: separate type

File: packages/frontend/src/services/http.service.ts (hunk 1 above)
Comment (IvanRuskevych) [reply packages/frontend/src/services/http.service.ts:59]: done

File: packages/frontend/src/shared/components/Layout/Layout.tsx (hunk 2 above)
Comment (IvanRuskevych) [reply packages/frontend/src/shared/components/Layout/Layout.tsx:26]: done

File: packages/frontend/src/shared/components/VerifyEmail/VerifyEmail.tsx (hunk 3 above)
Comment (IvanRuskevych) [reply packages/frontend/src/shared/components/VerifyEmail/VerifyEmail.tsx:18]: done

File: packages/frontend/src/shared/components/VerifyEmail/VerifyEmail.tsx (hunk 4 above)
Comment (IvanRuskevych) [reply packages/frontend/src/shared/components/VerifyEmail/VerifyEmail.tsx:10]: done

File: packages/frontend/src/modules/AuthForms/RegisterForm.tsx (hunk 5 above)
Comment (IvanRuskevych) [reply packages/frontend/src/modules/AuthForms/RegisterForm.tsx:17]: done

File: packages/frontend/src/modules/AuthForms/RegisterForm.tsx (hunk 6 above)
Comment (IvanRuskevych) [reply packages/frontend/src/modules/AuthForms/RegisterForm.tsx:21]: done

**Pull request merged**